### ISSUE TITLE
axios response interceptor will be triggered twice

### ISSUE THREAD

**Issue by yuu2lee4** [ctimmerm/axios-mock-adapter#61]
when i use passThrough feature，the passThrough request will trigger axios response interceptor twice.

**Comment by yuu2lee4** [comment]
It is ok after i use axios self not a instance created by axios when initialing a  mockAdapter.

**Comment by noru** [comment]
+1

**Comment by rickysang** [comment]
+1

**Comment by epaezrubio** [comment]
Any updates on this issue?

**Comment by epaezrubio** [comment]
This test reproduces the issue. Paste it into **pass_through.spec.js** and see how the data gets concatenated several times.

```javascript
  it('handles request transformations properly', function() {
    mock.onGet('/foo').passThrough();

    return instance.get('/foo', {
      data: 'foo',
      transformRequest: [function(data) {
        return data + 'foo';
      }]
    })
      .then(function(response) {
        expect(response.config.data).to.equal('foofoo');
      });
  });

  it('handles response transformations properly', function() {
    mock.onGet('/foo').passThrough();

    return instance.get('/foo', {
      transformResponse: [function(data) {
        return data + 'foo';
      }]
    })
      .then(function(response) {
        expect(response.data).to.equal('foo');
      });
  });
```

**Comment by wuzhibo0902** [comment]
> It is ok after i use axios self not a instance created by axios when initialing a mockAdapter.

But It's also call twice when I use axios itself not a instance.

**Comment by wuzhibo0902** [comment]
@yuu2lee4 It's also call twice when I use axios itself not a instance. Can you share your code?

**Comment by wuzhibo0902** [comment]
> It is ok after i use axios self not a instance created by axios when initialing a mockAdapter.

@yuu2lee4 It's also call twice when I use axios itself not a instance. Can you share your code?

**Comment by gtarsia** [comment]
I created an MR for this: https://github.com/ctimmerm/axios-mock-adapter/pull/189
You can reference this package with `yarn add gtarsia/axios-mock-adapter#master` or `npm i gtarsia/axios-mock-adapter#master`. It's an ugly hack, but it works. It's open to feedback.

**Comment by gtarsia** [comment]
I tried to fix it, but I couldn't really figure it out, so I switched to `fetch`, and coded the axios features I was unknowingly using, like automatic json conversion and query-string handling...

**Comment by dingziqi** [comment]
I fixed the problem at  #200

**Comment by damianobarbati** [comment]
+1
